### ISSUE TITLE
Add ability to reuse encoder for multiple files

### ISSUE THREAD

**Issue by tomwidmer** [chris-rudmin/opus-recorder#169]
Currently, the encoder can only be used to create a single ogg file. Then the web-worker is closed, and a new one created if you start recording again. It would be useful if the same web-worker could be reused to eliminate the overhead of recreating the web worker (including the wasm loading, etc.).

I think it's just a matter of destroying the old encoder (using _opus_encoder_destroy) then creating a new OggOpusEncoder (optionally with different/additional config?).

I'm looking into creating a PR for this.

**Comment by chris-rudmin** [comment]
@tomwidmer Amazing! I would love to see these changes as a set of Pull Requests. Small PRs would be best. Thanks!

**Comment by tomwidmer** [comment]
Will it be possible to merge the changes to master and release them soon?
Thanks

**Comment by chris-rudmin** [comment]
Are your changes complete?  I will test them out in their entirety and do a release if it looks good. :+1:

**Comment by tomwidmer** [comment]
Yes, all complete (just the 3 features, the flushing pause, encoded sample position and reuse). Thanks for your efforts in reviewing and testing!

**Comment by Ajaxy** [comment]
What about the same technique for decoder/wav workers? It seems that now they are re-created every time.
